lib/handler.ex

```diff
@@ -109,12 +133,20 @@ defmodule SAXMap.Handler do
 
   defp append_characters_text_content(nil, chars), do: chars
   defp append_characters_text_content(content, chars) when is_list(content) do
-    case Enum.reverse(content) do
-      [%{@key_text_content => text_items} | rest] ->
-        Enum.reverse([%{@key_text_content => [chars | text_items]} | rest])
-      items ->
-        Enum.reverse([%{@key_text_content => [chars]} | items])
-    end
+    # Optimized version - avoid double reversal by working with the list directly
+    append_characters_to_content(content, chars, [])
+  end
+
+  # Helper function to append characters without double reversal
+  defp append_characters_to_content([], chars, acc) do
+    [%{@key_text_content => [chars]} | acc]
+  end
+  defp append_characters_to_content([%{@key_text_content => text_items} | rest], chars, acc) do
+    # Found text content at head, prepend char and rebuild list
+    [%{@key_text_content => [chars | text_items]} | rest] ++ Enum.reverse(acc)
```

Comment:
```suggestion
    [%{@key_text_content => [chars | text_items]} | rest] ++ acc
```

With the previous mentioned comment, no need to reverse `acc`.